datastore/service.go

```diff
@@ -37,19 +39,30 @@ func (srv Service) GetAll(ctx context.Context, namespace models.NamespaceSpec, d
 func (srv Service) CreateResource(ctx context.Context, namespace models.NamespaceSpec, resourceSpecs []models.ResourceSpec, obs progress.Observer) error {
 	runner := parallel.NewRunner(parallel.WithLimit(ConcurrentLimit), parallel.WithTicket(ConcurrentTicketPerSec))
 	for _, resourceSpec := range resourceSpecs {
-		currentSpec := resourceSpec
-		repo := srv.resourceRepoFactory.New(namespace, currentSpec.Datastore)
+		incomingSpec := resourceSpec
+		repo := srv.resourceRepoFactory.New(namespace, incomingSpec.Datastore)
 		runner.Add(func() (interface{}, error) {
-			if err := repo.Save(ctx, currentSpec); err != nil {
+			proceed, err := srv.isProceedToSave(ctx, repo, incomingSpec)
+			if err != nil {
 				return nil, err
 			}
 
-			err := currentSpec.Datastore.CreateResource(ctx, models.CreateResourceRequest{
-				Resource: currentSpec,
+			if !proceed {
+				srv.notifyProgress(obs, &EventResourceCreated{
```

Comment:
the event will be different if skipped update

Reply:
updated by creating a new event